woocommerce/google-listings-and-ads
# Add WooCommerce Brands integration

### Changes proposed in this Pull Request:

As part of #16.

This PR introduces a new "Brand" attribute for products plus integration with the WooCommerce Brands extension. The brand attribute is displayed as a text input on the edit product page under the Google Listings and Ads section of product data.

Note: Brand can only be set for the parent product in case it's a variable product and NOT for the variations.

Since WooCommerce Brands allows the users to select multiple brands per product (?) and Google only expects one brand for each product we use the first selected brand as the primary brand for each product if multiple values are selected in WooCommerce Brands.

Another way to resolve the multiple brands issue would be to have the user define a "primary" brand (Similar to how Yoast SEO extension works) but I didn't go down that route, for now, to keep it simple. It might be worth exploring this option in the next versions. We could also implement this as part of the integration with the Yoast WooCommerce SEO extension but I doubt that's a priority right now.

### Detailed test instructions:

1. Make sure you have the WooCommerce Brands extension set up and enabled
2. Edit a product or create a new one
3. Assign a brand to the product using the WooCommerce Brand extension
4. Select the "From WooCommerce Brand" for the brand attribute edit product page (under the Google Listings and Ads section of product data)
5. Publish or update the product and wait for it to sync (or sync manually via the Connection Test page)
6. Check Google Merchant Center and make sure the brand attribute is set correctly for the product
7. Repeat the above steps for a variable product and make sure the brand is set correctly (the same value) or all variations
